### PR TITLE
actually disable grpc when no grpc & fixes grpc issues on linux

### PR DESCRIPTION
**Pull request by fundies**
partner of https://github.com/enigma-dev/enigma-dev/pull/1919

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,10 @@
 cmake_minimum_required(VERSION 3.11)
 
-include(CMakeDependentOption)
-cmake_dependent_option(RGM_ENABLE_GRPC_SERVER "Enable the GRPC client plugin for compilation and code analysis." ON "WIN32" OFF)
+if (WIN32)
+  option(RGM_ENABLE_GRPC_SERVER "Enable the GRPC client plugin for compilation and code analysis." ON)
+else()
+  option(RGM_ENABLE_GRPC_SERVER "Enable the GRPC client plugin for compilation and code analysis." OFF)
+endif()
```
Comment (RobertBColton) [inline CMakeLists.txt]: Why did you do this?

File: MainWindow.cpp [hunk 2]
```diff
@@ -20,7 +20,10 @@
 #include "Components/Logger.h"
 
 #include "Plugins/RGMPlugin.h"
+
+#ifdef RGM_SERVER_ENABLED
 #include "Plugins/ServerPlugin.h"
+#endif
```
Comment (RobertBColton) [inline MainWindow.cpp]: Thanks, I did miss that one.

File: CMakeLists.txt (hunk 1 above)
Comment (fundies) [reply CMakeLists.txt]: cause your way completely hides the option on linux mine just disables it

Comment (fundies) [reply CMakeLists.txt]: also, cause i dont have OCD like u

Comment (RobertBColton) [reply CMakeLists.txt]: Well it's not really OCD, it's just like if you have to type that string multiple times you are more likely to make a typo and cause a run time bug. Whereas if you make a typo in my way, even though it's more characters, it's instead more likely to be a compile time error.

Comment (fundies) [reply CMakeLists.txt]: sounds like ocd to me. if you tested that works ill append the pr tho

File: CMakeLists.txt [hunk 3]
```diff
@@ -317,4 +317,4 @@ if (WIN32)
           fixup_bundle(\"${APPS}\" \"${LIBS}\" \"${SEARCH_PATHS}\")
           execute_process(COMMAND windeployqt.exe ${WINDEPLOY_ARGS} ${APPS})
           ")
-endif()
+endif()
```
Comment (RobertBColton) [inline CMakeLists.txt]: Missed end of line!

Comment (RobertBColton) [reply CMakeLists.txt]: I got it.

**Pull request merged**